scale/job/models.py

```diff
@@ -2435,53 +2435,45 @@ def create_job_type_v5(self, name, version, interface, trigger_rule=None, error_
         return job_type
 
     @transaction.atomic
-    def create_job_type_v6(self, docker_image, manifest_dict, icon_code=None, max_scheduled=None,  
-                            configuration_dict=None):
+    def create_job_type_v6(self, docker_image, manifest, icon_code=None, max_scheduled=None,  
+                            configuration=None, secrets=None):
         """Creates a new Seed job type and saves it in the database. All database changes occur in an atomic
         transaction.
 
         :param docker_image: The docker image containing the code to run for this job.  
         :type docker_image: string
-        :param manifest_dict: The Seed Manifest defining the interface for running a job of this type
-        :type manifest_dict: dict
+        :param manifest: The Seed Manifest defining the interface for running a job of this type
+        :type manifest: :class:`job.seed.manifest.SeedManifest`
         :param icon_code: A font-awesome icon code to use when representing this job type.
         :type icon_code: string
         :param max_scheduled: Maximum  number of jobs of this type that may be scheduled to run at the same time.
         :type max_scheduled: integer
-        :param configuration_dict: The configuration for running a job of this type, possibly None
-        :type configuration_dict: dict
+        :param configuration: The configuration for running a job of this type, possibly None
+        :type configuration: :class:`job.configuration.configuration.JobConfiguration`
+        :param secrets: Secret settings required by this job type
+        :type secrets: dict
         :returns: The new job type
         :rtype: :class:`job.models.JobType`
 
         :raises :class:`job.exceptions.InvalidJobField`: If a given job type field has an invalid value
         """
 
-        # Create manifest and validate
-        manifest = SeedManifest(manifest_dict)
-
         # Create/update any errors defined in manifest
         error_mapping = manifest.get_error_mapping()
         error_mapping.save_models()
 
         # Create the new job type
         job_type = JobType()
-        
-        secrets = None
-        if configuration_dict:
-            configuration = JobConfigurationV6(configuration_dict, do_validate=True).get_configuration()
-            configuration.validate(manifest)
-            secrets = configuration.remove_secret_settings(manifest)
-            configuration_dict = convert_config_to_v6_json(configuration).get_dict()
-            job_type.configuration = configuration_dict
```

Comment:
Sorry if I wasn't clear. My intent was for the view to have just the line converting the v6 JSON:
`configuration = JobConfigurationV6(configuration_dict, do_validate=True).get_configuration()`

This method would then contain the rest of the code, something like this:
```
if not configuration:
    configuration = JobConfiguration()
configuration.validate(manifest)
secrets = configuration.remove_secret_settings(manifest)
job_type.configuration = convert_config_to_v6_json(configuration).get_dict()

```

This way the view is handling the JSON validation logic with dealing with user input and the manager method is handling business logic/validation and operates on the logic class (`JobConfiguration`).